tests/gnrc_sock_tcp/main.c

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (C) 2021   Simon Brummer <simon.brummer@posteo.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "shell.h"
+#include "msg.h"
+#include "net/sock/tcp.h"
+
+#define MAIN_QUEUE_SIZE (8)
+#define SOCK_TCP_QUEUE_SIZE (1)
+#define BUFFER_SIZE (1024)
+
+static msg_t main_msg_queue[MAIN_QUEUE_SIZE];
+static sock_tcp_t socks[SOCK_TCP_QUEUE_SIZE];
+static sock_tcp_t *sock = socks;
+static sock_tcp_queue_t sock_queue;
+static char buffer[BUFFER_SIZE];
+
+void dump_args(int argc, char **argv)
+{
+    printf("%s: ", argv[0]);
+    printf("argc=%d", argc);
+    for (int i = 0; i < argc; ++i) {
+        printf(", argv[%d] = %s", i, argv[i]);
+    }
+    printf("\n");
+}
+
+int sock_tcp_connect_cmd(int argc, char **argv)
+{
+    dump_args(argc, argv);
+
+    sock_tcp_ep_t ep = SOCK_IPV6_EP_ANY;
+    gnrc_tcp_ep_from_str((gnrc_tcp_ep_t *) &ep, argv[1]);
+    uint16_t local_port = atol(argv[2]);
+    uint16_t flags = 0;
+
+    int err = sock_tcp_connect(sock, &ep, local_port, flags);
+    switch (err) {
+        case -EADDRINUSE:
+            printf("%s: returns -EADDRINUSE\n", argv[0]);
+            break;
+
+        case -EAFNOSUPPORT:
+            printf("%s: returns -EAFNOSUPPORT\n", argv[0]);
+            break;
+
+        case -ECONNREFUSED:
+            printf("%s: returns -ECONNREFUSED\n", argv[0]);
+            break;
+
+        case -EINVAL:
+            printf("%s: returns -EINVAL\n", argv[0]);
+            break;
+
+        case -ENETUNREACH:
+            printf("%s: returns -ENETUNREACH\n", argv[0]);
+            break;
+
+        case -ENOMEM:
+            printf("%s: returns -ENOMEM\n", argv[0]);
+            break;
+
+        case -EPERM:
+            printf("%s: returns -EPERM\n", argv[0]);
+            break;
+
+        case -ETIMEDOUT:
+            printf("%s: returns -ETIMEDOUT\n", argv[0]);
+            break;
```

Comment:
Will do

Reply:
Are those supported by all boards (if yes some of my worries for #16705 will be lifted as well ;-))

Reply:
Looks like only AVR is missing this function - I'd rather add `strerror()` to `cpu/avr8_common/avr_libc_extra` then instead of replicating it in-line.

Reply:
I've added strerror now. I move the error code printing to a dedicated function because calls to strerror return different values for 0 on different platforms and the python script matches for this string.